airbyte-integrations/connectors/source-medallia/source_medallia/source.py

```diff
@@ -0,0 +1,271 @@
+            stream_state: Mapping[str, Any],
+            stream_slice: Optional[Mapping[str, Any]] = None,
+            next_page_token: Optional[Mapping[str, Any]] = None,
+    ) -> Optional[Mapping]:
+        op = sgqlc.operation.Operation(_schema_root.query_type)
+
+        order = {"fieldId": "a_initial_finish_timestamp", "direction": "ASC"}
+
+        filter = {"fieldIds": ["a_initial_finish_timestamp"], "gt": str(self._cursor_value)}
+
+        var = _schema.RecordOrder(order)
+
+        # do not apply after, will have no pages left when all data loaded
+        feedback = op.feedback(first=self.limit, order_by=[var], filter=filter)
+        feedback.nodes.id()
+
+        extend_field_data = sgqlc.operation.Fragment(_schema_root.FieldData, 'extendFieldData')
+
+        extend_field_data.__as__(_schema_root.EnumFieldData).options()
+        extend_field_data.__as__(_schema_root.StringFieldData).values()
+
+        extend_field_data.__as__(_schema_root.CommentFieldData).rule_topic_taggings_page(__alias__='topics').nodes()
+        extend_field_data.__as__(_schema_root.CommentFieldData).data_topic_taggings_page(__alias__='themes').nodes()
+        extend_field_data.__as__(_schema_root.CommentFieldData).sentiment_taggings_page(__alias__='sentiment').nodes()
+
+        extend_field_data.__as__(_schema_root.IntFieldData).values()
+        extend_field_data.__as__(_schema_root.DateFieldData).values()
+        extend_field_data.__as__(_schema_root.UnitFieldData).units().id()
+
+        meta_data = feedback.nodes.field_data_list(__alias__='metaData',
+                                                   filter_unanswered=True,
+                                                   field_ids=self.metadata_columns)
+
+        meta_data.field().id()
+        meta_data.field().name()
+
+        meta_data.__fragment__(extend_field_data)
+
+        question_data = feedback.nodes.field_data_list(__alias__='questionData',
+                                                       filter_unanswered=True,
+                                                       field_ids=self.question_columns)
+
+        question_data.field().id()
+        question_data.field().name()
+
+        question_data.__fragment__(extend_field_data)
+
+        feedback.page_info()
+        feedback.page_info.has_next_page()
+        feedback.page_info.end_cursor()
+
+        return {'query': str(op)}
+
+
+class SourceMedallia(AbstractSource):
+
+    def get_fields(self, config: json):
+
+        initialization_params = {"authenticator": initialize_authenticator(config), "url_base": config.get("query-endpoint")}
+
+        field_id = FieldId(**initialization_params)
+
+        data = read_full_refresh(field_id)
+
+        nodes = {}
+
+        for node in data:
+            nodes[node["id"]] = node
+
+        metadata_columns = list(filter(lambda x: x.startswith(tuple(['a_', 'u_', 'e_', 'k_'])), nodes))
+        question_columns = list(filter(lambda x: x.startswith('q_'), nodes))
```

Comment:
Can you create constant tuple like METADATA_PREFIX_COLUMN and QUESTION_PREFIX_COLUMN?